lib/src/components/NxTable/NxTable.tsx

```diff
@@ -17,18 +17,25 @@ import { ColumnCountContext } from './contexts';
 import { NxTableProps, nxTablePropTypes } from './types';
 export { NxTableProps };
 
+const mutationObserverConfig = { subtree: true, childList: true, attributes: false, characterData: false };
+
 const NxTable = function NxTableElement(props: NxTableProps) {
   const {className, children, ...attrs} = props,
-      thead = only(children, NxTableHead),
-      tbody = only(children, NxTableBody),
-      trow = thead && only(thead.props.children, NxTableRow),
-      columns = trow ? React.Children.count(trow.props.children) : 0;
+      tableRef = useRef<HTMLTableElement>(null),
+      [columnCount, setColumnCount] = useState(1),
+      updateColumnCount = useCallback(function updateColumnCount() {
+        if (tableRef.current) {
+          setColumnCount(tableRef.current.querySelectorAll('thead > tr:first-child > th').length);
```

Comment:
> It occurs to me that this could be incorrect if the first header row contains cells that have a colspan value set. One option would be to scan all the table rows and use the max number of cells in any particular row for the column count.

I'm not really inclined to scan the whole table for performance reasons.  We could maybe look for and add up `colspan` attributes in the first row.